src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -211,6 +332,91 @@ private IFeatureFilterMetadata GetFeatureFilterMetadata(string filterName)
+                        }
+
+                        return IsMatchingMetadataName(name, assignerName, assignerSuffix);
+                    });
+
+                    if (matchingAssigners.Count() > 1)
+                    {
+                        throw new FeatureManagementException(FeatureManagementError.AmbiguousFeatureVariantAssigner, $"Multiple feature variant assigners match the configured assigner named '{assignerName}'.");
+                    }
+
+                    return matchingAssigners.FirstOrDefault();
+                }
+            );
+
+            return assigner;
+        }
+
+        /// <summary>
+        /// Trims a suffix from a name if necessary
+        /// </summary>
+        /// <param name="name">The name to trim.</param>
+        /// <param name="suffix">The possible suffix that may need trimming.</param>
+        /// <returns></returns>
+        private static string GetTrimmedName(string name, string suffix)
+        {
+            Debug.Assert(name != null);
+            Debug.Assert(suffix != null);
+
+            if (name.EndsWith(suffix, StringComparison.OrdinalIgnoreCase))
+            {
+                name = name.Substring(0, name.Length - suffix.Length);
+            }
+
+            return name;
+        }
+
+        private static bool IsMatchingMetadataName(string metadataName, string desiredName, string suffix)
+        {
+            //
+            // Support matching with suffix, i.e. "CustomFilter"
+            // Or without, i.e. "Custom"
+            return desiredName.EndsWith(suffix, StringComparison.OrdinalIgnoreCase) ?
+                IsMatchingMetadataName(metadataName, desiredName) :
+                IsMatchingMetadataName(GetTrimmedName(metadataName, suffix), desiredName);
+        }
+
+        private static bool IsMatchingMetadataName(string metadataName, string desiredName)
+        {
+            //
+            // Feature filters can have namespaces in their alias
+            // If a feature is configured to use a filter without a namespace such as 'MyFilter', then it can match 'MyOrg.MyProduct.MyFilter' or simply 'MyFilter'
+            // If a feature is configured to use a filter with a namespace such as 'MyOrg.MyProduct.MyFilter' then it can only match 'MyOrg.MyProduct.MyFilter' 
+            if (desiredName.Contains('.'))
+            {
+                //
+                // The configured filter name is namespaced. It must be an exact match.
+                return string.Equals(metadataName, desiredName, StringComparison.OrdinalIgnoreCase);
+            }
+            else
+            {
+                //
+                // We take the simple name of a filter, E.g. 'MyFilter' for 'MyOrg.MyProduct.MyFilter'
+                string simpleName = metadataName.Contains('.') ? metadataName.Split('.').Last() : metadataName;
+
+                return string.Equals(simpleName, desiredName, StringComparison.OrdinalIgnoreCase);
```

Comment:
I think we can just add below before this line (the return). We don't need the other IsMatchingMetadataName overload.

```cs
if (!desiredName.EndsWith(suffix, StringComparison.OrdinalIgnoreCase))
{
    simpleName = GetTrimmedName(simpleName, suffix);
}

```